Neos.Neos/Classes/Fusion/AbstractMenuItemsImplementation.php

```diff
@@ -60,6 +60,13 @@ abstract class AbstractMenuItemsImplementation extends AbstractFusionObject
      */
     protected $renderHiddenInIndex;
 
+    /**
+     * Internal cache for the calculateItemStates property.
+     *
+     * @var boolean
+     */
+    protected $calculateItemStates;
```

Comment:
nitpick, a little more defensive:
```suggestion
     */
    private ?bool $calculateItemStates = null;
```

Reply:
why not directly 
```
private bool $calculateItemStates = false;
```

Reply:
because you just use this field as runtime cache for `isCalculateItemStatesEnabled()` and use `null` to distinguish the cached from the uncached case, remember? :)
I started suffixing those fields with some `...RuntimeCache` to make that clearer.

I'd say it's OK to introduce private fields even if we still use `protected`, especially if it is just an internal field with a dedicated accessor (like here). But it was just a nitpick anyways

Reply:
yes its more of a drop of water on a hot stone? I would refrain from discussing such minor things 😅